### PR TITLE
add candlestick chart to pwa sample app

### PR DESCRIPTION
**Pull request by marin-bratanov**
caveats:
* no format/template for axis labels - cannot get currency symbol on the y-axis
* no smooth style for navigator area series
* on Refresh() (browser resize) the to/from sliders go back to initial position since there is no two-way binding

### CODE REVIEW

**Comment by Stamo-Gochev** [comment]
Adding these features requires a new version of Telerik UI for Blazor, let me know if this blocks the update of the app.

The options can be set using JavaScript with the [setOptions](https://github.com/telerik/blazor/blob/master/js/telerik-blazor/src/charts/chart.js#L77) method meanwhile.

**Comment by matanasov** [comment]
@Stamo-Gochev as far as I see the app is already updated to 2.16.0. What new version do you mean?

@marin-bratanov Can you add crosshairs tooltips because without them, they are not very useful. Or just remove the crosshairs.

**Comment by Stamo-Gochev** [comment]
> 
> 
> @Stamo-Gochev as far as I see the app is already updated to 2.16.0. What new version do you mean?

Adding the features listed in https://github.com/telerik/blazor-ui/pull/47#issue-464782035 require an implementation and a new version of Telerik UI for Blazor to be released, so that they are officially available.

**Comment by marin-bratanov** [comment]
Crosshair tooltips added, they reduce the performance significantly, and aren't present in the react implementation (which is why I didn't add them).

Data generation was improved to ensure High > Open and Low < Close.

File: sample-applications/blazor-stocks/Client/Components/StocksChart/MainChart.razor [hunk 1]
```diff
@@ -6,46 +6,128 @@
...
+        </StockChartValueAxes>
+
+        <StockChartLegend Visible="false" />
+        <StockChartTooltip Visible="true" />
+
+    </TelerikStockChart>
+}
+else
+{
+    <TelerikChart Transitions="true" RenderAs="@RenderingMode.SVG" @ref="@ChartRef">
```
Comment (Stamo-Gochev) [inline sample-applications/blazor-stocks/Client/Components/StocksChart/MainChart.razor:87]: This is the default values for `Transitions` and `RenderAs` - why they are listed explicitly?

Comment (marin-bratanov) [reply sample-applications/blazor-stocks/Client/Components/StocksChart/MainChart.razor]: That is a great question, to which I have no reasonable answer :) https://github.com/telerik/blazor-ui/pull/47/commits/45ec0ec2e5d33a117dee3d87295b5582a815feba

**Comment by marin-bratanov** [comment]
The issue was that the random data generation was, indeed, random, so sometimes Low values were lower than High values and things like that. I think I fixed it by ensuring that the High and Low are always the highest and lowest value in the model.

**Pull request merged**